jobinfo

```diff
@@ -179,6 +179,106 @@ def f_state(x, meta):
         deps = " (%s)" % deps
     return ','.join(states) + reason + deps
 
+def memory_to_bytes(memory_spec, cores_per_node):
+   # Find unit of memory request
+   unit=memory_spec[-2:-1]
+   # Find number in memory request
+   memory_req=float(memory_spec[:-2])
+   # Transform the value into bytes
+   if unit == 'K':
+      memory_req = memory_req*1024
+   elif unit == 'M':
+      memory_req = memory_req*1024**2
+   elif unit == 'G':
+      memory_req = memory_req*1024**3
+   elif unit == 'T':
+      memory_req = memory_req*1024**4
+   elif unit == 'P':
+      memory_req = memory_req*1024**5
+   # If the memory request was per core, multiply by the average number of cores
+   # This is the best we can do given the lack of detailed information
+   if memory_spec[-1] == 'c':
+      memory_req = memory_req*cores_per_node
+   return memory_req
+
+def get_cpus_node(nodenames):
+    nodename=nodenames.split(',')[0]
+    info = subprocess.Popen(['scontrol', 'show','node', nodename],stdout=subprocess.PIPE)
+    for line in info.stdout:
+        if 'CfgTRES' in line:
+            cpucount = re.search('cpu=(\d+)',line).group(1)
+    return cpucount
+
+def get_hints(meta):
+    # Parameters affecting the behaviour of this routine
+    min_walltime = 180          # Minimum walltime needed before printing job hints
+    min_memory = 2*1024**3      # Minimum request that will be ignored
+    ignore_partitions = ['gpu'] # Partitions for which to ignore memory and cpu usage, e.g. GPU partitions
```

Comment:
These settings are a bit hidden and hard to modify now. Maybe we should put them in a configuration file at some point (when we want to have more of these tuneable parameters), or move them to the top of this file, but I guess this will do for now?

Reply:
Moved the parameters to the start of the script.